### PR TITLE
Update links to the provider documentation in docs and comments.

### LINKED ISSUE #13506: Wrong redirection in documentation
I noticed wrong redirect on pages, where redirection point to `https://www.terraform.io/docs/providers/azurerm/index.html` instead of `https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs` 

Example pages with wrong redirect (look at redirect in `features setting within the Provider block`):
* [azurerm_linux_virtual_machine](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/linux_virtual_machine)
* [azurerm_linux_virtual_machine_scale_set](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/linux_virtual_machine_scale_set)
* [azurerm_windows_virtual_machine](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/windows_virtual_machine)

Page with well working redirection:
* [azurerm_resource_group](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/resource_group)

I think it would be good to correct it.

### PR DESCRIPTION
**Pull request by atombrella**
Fixes #13506

### CODE REVIEW

File: website/docs/r/linux_virtual_machine.html.markdown [hunk 1]
```diff
@@ -12,7 +12,8 @@ Manages a Linux Virtual Machine.
 
 ## Disclaimers
 
--> **Note** Terraform will automatically remove the OS Disk by default - this behaviour can be configured [using the `features` setting within the Provider block](https://www.terraform.io/docs/providers/azurerm/index.html#argument-reference).
+-> **Note** Terraform will automatically remove the OS Disk by default - this behaviour can be configured [using the `features` setting within the Provider block](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs#skip_provider_registration
```
Comment (stephybun) [inline website/docs/r/linux_virtual_machine.html.markdown:15]: From the context of the sentence the link should actually point to the features block
```suggestion
-> **Note** Terraform will automatically remove the OS Disk by default - this behaviour can be configured [using the `features` setting within the Provider block](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs#features
```

File: website/docs/r/linux_virtual_machine_scale_set.html.markdown [hunk 2]
```diff
@@ -14,7 +14,7 @@ Manages a Linux Virtual Machine Scale Set.
 
 ~> **Note:** All arguments including the administrator login and password will be stored in the raw state as plain-text. [Read more about sensitive data in state](/docs/state/sensitive-data.html).
 
--> **Note:** Terraform will automatically update & reimage the nodes in the Scale Set (if Required) during an Update - this behaviour can be configured [using the `features` setting within the Provider block](https://www.terraform.io/docs/providers/azurerm/index.html#features).
+-> **Note:** Terraform will automatically update & reimage the nodes in the Scale Set (if Required) during an Update - this behaviour can be configured [using the `features` setting within the Provider block](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs#skip_provider_registration#features).
```
Comment (stephybun) [inline website/docs/r/linux_virtual_machine_scale_set.html.markdown:17]: Same here

File: website/docs/r/windows_virtual_machine.html.markdown [hunk 3]
```diff
@@ -12,7 +12,8 @@ Manages a Windows Virtual Machine.
 
 ## Disclaimers
 
--> **Note** Terraform will automatically remove the OS Disk by default - this behaviour can be configured [using the `features` setting within the Provider block](https://www.terraform.io/docs/providers/azurerm/index.html#argument-reference).
+-> **Note** Terraform will automatically remove the OS Disk by default - this behaviour can be configured [using the `features` setting within the Provider block](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs#skip_provider_registration
```
Comment (stephybun) [inline website/docs/r/windows_virtual_machine.html.markdown:15]: Same here

**Review by stephybun — CHANGES_REQUESTED** [review]
Thanks for this PR @atombrella, a few of the links point to the `skip_provider_registration` property instead of the `features` block. Once those are fixed this LGTM 👍🏼

File: website/docs/r/windows_virtual_machine.html.markdown (hunk 3 above)
Comment (atombrella) [reply website/docs/r/windows_virtual_machine.html.markdown]: Both are updated. Thanks :)

**Review by stephybun — APPROVED** [review]

**Comment by stephybun** [comment]
Thanks @atombrella, LGTM!

**Pull request merged**